checkov/terraform/plan_parser.py

```diff
@@ -46,76 +49,94 @@ def _hclify(obj, parent_key=None):
 
         if _is_list_of_dicts(value):
             child_list = []
-            for internal_val in value:
-                child_list.append(_hclify(internal_val))
+            conf_val = conf.get(key, []) if conf else []
+            for internal_val, internal_conf_val in itertools.zip_longest(value, conf_val):
+                child_list.append(_hclify(internal_val, internal_conf_val))
             ret_dict[key] = child_list
         if isinstance(value, dict):
-            child_dict = _hclify(value, key)
+            child_dict = _hclify(value, parent_key=key)
             if parent_key == "tags":
                 ret_dict[key] = child_dict
             else:
                 ret_dict[key] = [child_dict]
+    if conf:
+        for conf_key in conf.keys() - obj.keys():
+            ref = next((x for x in conf[conf_key].get("references", []) if not x.startswith(("var.", "local."))), None)
+            if ref:
+                ret_dict[conf_key] = [ref]
+
     return ret_dict
 
-def _prepare_resource_block(resource):
+
+def _prepare_resource_block(resource: dict_node, conf: Optional[dict_node]) -> Tuple[Dict[str, Dict[str, Any]], bool]:
```

Comment:
@gruebel Can you please add `conf` to the docs in the next lines, explaining what it is and what it is used for?